webpack/assets/javascripts/react_app/components/common/forms/CounterInput/CounterInput.js

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect, useState } from 'react';
+import RCInputNumber from 'rc-input-number';
+import PropTypes from 'prop-types';
+import { translate as __ } from '../../../../common/I18n';
+import { noop } from '../../../../common/helpers';
+
+const CounterInput = ({
+  value,
+  recommendedMaxValue,
+  maxValue,
+  onChange,
+  minValue,
+  disabled,
+  name,
+  id,
+  setError,
+  setWarning,
+}) => {
+  const [innerValue, setInnerValue] = useState(value);
+  useEffect(() => {
+    if (maxValue && innerValue > maxValue) {
+      setWarning(null);
+      setError(__('Specified value is higher than maximum value'));
+    } else if (recommendedMaxValue && innerValue > recommendedMaxValue) {
+      setError(null);
+      setWarning(__('Specified value is higher than recommended maximum'));
+    } else {
+      setError(null);
+      setWarning(null);
+    }
+  }, [recommendedMaxValue, maxValue, innerValue, setError, setWarning]);
```

Comment:
we can ommit the `setState` (setError, setWarning) setters from the dependency list, react guarantees that `setState` is stable during renders.

https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies
https://atomizedobjects.com/blog/react/how-to-use-setstate-from-within-useeffect-with-dependencies/